src/ripple/app/main/Main.cpp

```diff
@@ -488,6 +495,81 @@ int run (int argc, char** argv)
             crypto_prng().load_state(entropy.string ());
     }
 
+    if (vm.count("vacuum"))
+    {
+        DatabaseCon::Setup dbSetup = setup_DatabaseCon(*config);
+        if (dbSetup.standAlone)
+        {
+            std::cerr << "vacuum not applicable in standalone mode.\n";
+            return -1;
+        }
+        boost::filesystem::path dbPath = dbSetup.dataDir / TxnDBName;
+        auto txnDB = std::make_unique<DatabaseCon> (dbSetup, TxnDBName,
+            TxnDBInit, TxnDBCount);
+        if (txnDB.get() == nullptr)
+        {
+            std::cerr << "Cannot create connection to " << dbPath.string() <<
+                '\n';
+            return -1;
+        }
+        boost::system::error_code ec;
+        uintmax_t dbSize = boost::filesystem::file_size(dbPath, ec);
+        if (ec)
+        {
+            std::cerr << "Error checking size of " << dbPath.string() << ": "
+                << ec.message() << '\n';
+            return -1;
+        }
+
+        assert(dbSize != static_cast<uintmax_t>(-1));
+
+        std::string tmpDir = vm["vacuum"].as<std::string>();
+        boost::filesystem::path tmpPath = tmpDir;
+        if (boost::filesystem::space(tmpPath, ec).available < dbSize)
+        {
+            if (ec)
+            {
+                std::cerr << "Error checking status of " << tmpPath.string()
+                    << ": " << ec.message() << '\n';
+            }
+            else
+            {
+                std::cerr << "A valid directory for vacuuming must be "
+                             "specified on a filesystem with at least "
+                             "as much free space as the size of " <<
+                             dbPath.string() << ", which is " <<
+                             dbSize << " bytes. The filesystem for " <<
+                             tmpPath.string() << " only has " <<
+                             boost::filesystem::space(tmpPath, ec).available
+                             << " bytes.\n";
+            }
+            return -1;
+        }
+
+        auto db = txnDB->checkoutDb();
+        std::uint32_t pageSize;
+        try
+        {
+            *db << "PRAGMA page_size;", soci::into(pageSize);
+            std::cout << "VACUUM beginning. page_size: " << pageSize
+                << std::endl;
+            *db << "PRAGMA journal_mode=OFF;";
+            *db << "PRAGMA temp_store_directory=\"" << tmpPath.string()
+                << "\";";
+            *db << "VACUUM;";
+            *db << "PRAGMA journal_mode=WAL;";
+            *db << "PRAGMA page_size;", soci::into(pageSize);
+        }
+        catch (soci::soci_error const& e)
+        {
+            std::cerr << "SQLite error: " << e.what() << '\n';
+            return 1;
```

Comment:
I notice this return value is different than the other failure cases...it doesn't matter although I wondered if we impart meaning to exit values in this case.